server/types/callbacks.go

```diff
@@ -11,12 +11,10 @@ type ConnectionResponse struct {
 	Accept             bool
 	HTTPStatusCode     int
 	HTTPResponseHeader map[string]string
+	ConnectionHandler  ConnectionCallbacks
```

Comment:
Nit: should the field be also named ConnectionCallbacks? Or rename ConnectionCallbacks struct to ConnectionHandler?
The names being different somehow is not helping me to read.